moc_openshift.py

```diff
@@ -575,3 +619,60 @@ def update_rolebindings(self, project_name, role, rolebindings_json):
                 payload["metadata"][key] = rolebindings_json["metadata"][key]
         self.logger.debug("payload -> 2: " + json.dumps(payload))
         return self.put_request(url, payload, True)
+
+    # member functions for quotas
+    def generate_openshift_quota(self, project_name, quota_def):
+        payload = {
+            "apiVersion": "v1",
+            "kind": "ResourceQuota",
+            "metadata": {
+                "name": self.generate_quota_name(self, project_name, quota_def),
+                "namespace": project_name,
+            },
+            "spec": {},
+        }
+        for k in quota_def.keys():
+            if k == "scope":
+                payload["spec"]["scopes"] = [quota_def["scope"]]
+            else:
+                payload["spec"][k] = quota_def[k]
+        return payload
+
+    def generate_quota_def(self, payload):
+        scope_count = len(payload["spec"]["scopes"])
+        if scope_count == 0:
+            scope_count = 1
+        quota_def = list({} for i in range(scope_count))
+        for k in payload["spec"].keys():
```

Comment:
Rather than calling `range` and relying on a priori knowledge of the number of scopes in the payload, you could rewrite the function like this:

```
def generate_quota_def_new(payload):
    quota_def = []

    # Get a dictionary of "everything in payload other than
    # the 'scopes' key".
    quota_vals = {k: v for k, v in payload['spec'].items()
                  if k != 'scopes'}

    # Get a list of scopes, or default to a list with a single
    # scope named 'Global'.
    scopes = payload['spec'].get('scopes', []) or ['Global']
    for scope in scopes:
        qd = {'scope': scope}
        qd.update(copy.deepcopy(quota_vals))
        quota_def.append(qd)

    return quota_def
```

I suspect that you could also drop the use of `copy.deepcopy`, unless
you are further modifying the values for each quota definition
elsewhere in the code.